### PR TITLE
Cybersource Checkout Integration

### PR DESCRIPTION
**Pull request by esinx**
(Incomplete) Needs backend integration, but can be merged for frontend development

### CODE REVIEW

File: frontend/components/Tickets/CartTickets.tsx [hunk 1]
```diff
@@ -127,14 +127,41 @@ const combineTickets = (tickets: EventTicket[]): CountedEventTicket[] =>
...
+  const [captureContext, setCaptureContext] = useState<string>()
+
+  const fetchToken = async () => {
+    const res = await doApiRequest(`/tickets/initiate_checkout/?format=json`, {
+      method: 'POST',
+      body: null,
+    })
+    const data = await res.json()
+    if (!data.success) {
+      throw new Error(data.detail)
```
Comment (joyliu-q) [inline frontend/components/Tickets/CartTickets.tsx:141]: Let's toast here

File: frontend/components/Tickets/CartTickets.tsx [hunk 2]
```diff
@@ -334,8 +348,25 @@ const CartTickets: React.FC<CartTicketsProps> = ({ tickets, soldOut }) => {
...
+                '/tickets/complete_checkout/?format=json',
+                {
+                  method: 'POST',
+                  body: {
+                    transient_token: transientToken,
+                  },
+                },
+              )
+              const data = await res.json()
+              console.log({ data })
```
Comment (joyliu-q) [inline frontend/components/Tickets/CartTickets.tsx:366]: console.log

**Review by joyliu-q — COMMENTED** [review]
In cybersource I trust. Left some other comments related to nit stuff

File: frontend/components/Tickets/CartTickets.tsx (hunk 1 above)
Comment (esinx) [reply frontend/components/Tickets/CartTickets.tsx:141]: good suggestion

File: frontend/components/Tickets/CartTickets.tsx (hunk 2 above)
Comment (esinx) [reply frontend/components/Tickets/CartTickets.tsx:366]: yes

**Review by julianweng — COMMENTED** [review]
Thanks for your work Eunsoo! Couple of general comments:

- Checkout seems to work for me completely (using my own card), resulting in the event becoming owned. Is the "needs backend integration" just referring to the implementation of the final callback where the user isn't notified of success?

- This could be for next iteration, but would be nice to incorporate cart lock on checkout logic considerations into the user flow. Assuming capture contexts aren't invalidated on page refresh, if a user cart is already locked / has a capture context, the payment modal should be automatically opened using the saved capture context IMO, especially since the user isn't supposed to be able to renew the 10 minute holds by checking out again.

- Somewhat related: is the close button on the modal supposed to be enabled? Given the previous considerations, we would probably have to implement a prompt notifying the user that they are releasing their holds etc. by exiting, which can be separate from this PR, but just wanted to get your thoughts.

**Review by julianweng — APPROVED** [review]
LGTM, left a couple considerations: feel free to address the free tickets one on the associated feature branch after merging this into ticketing/the feature branch.

File: frontend/components/Tickets/CartTickets.tsx [hunk 3]
```diff
@@ -127,14 +128,61 @@ const combineTickets = (tickets: EventTicket[]): CountedEventTicket[] =>
...
+  const [captureContext, setCaptureContext] = useState<string>()
+
+  const fetchToken = async () => {
+    const res = await doApiRequest(`/tickets/initiate_checkout/?format=json`, {
+      method: 'POST',
+      body: null,
+    })
+    const data = await res.json()
+    if (!data.success) {
+      toast.error(data.detail, {
```
[the remark this answers is not in the document]
Comment (esinx) [reply frontend/components/Tickets/CartTickets.tsx:142]: Yeah I think we can merge this for now, let's create a new issue for checking for free tickets. For the demo, I believe we do want to show off that real payments can be made.

**Pull request merged**